### PR TITLE
[rootedtoon] Initial contribution

### PR DESCRIPTION
**Pull request by DaanMeijer**
[rootedtoon] add the binding to the main repository

This is the addition of the binding as discussed on https://community.openhab.org/t/rooted-locally-accessible-toon/65958. The binding connects to a rooted version of Toon, a dutch thermostat panel.

Built jar is downloadable here: https://gofile.io/?c=uPo8Kk

### CODE REVIEW

**Comment by 9037568** [comment]
I've marked this as a WIP since it isn't finished.

**Comment by DaanMeijer** [comment]
> I've marked this as a WIP since it isn't finished.

Is there something I can do to help this PR get more finished? Or is this a matter of waiting for the build to finish?

**Comment by 9037568** [comment]
>Is there something I can do to help this PR get more finished? Or is this a matter of waiting for the build to finish?

My statement was based on the fact that your README is entirely devoid of content.
When you've fixed that, if you're done with the rest of the coding, please go ahead and remove the [WIP] tag to show this is ready to be reviewed.

**Comment by DaanMeijer** [comment]
> > Is there something I can do to help this PR get more finished? Or is this a matter of waiting for the build to finish?
> 
> My statement was based on the fact that your README is entirely devoid of content.
> When you've fixed that, if you're done with the rest of the coding, please go ahead and remove the [WIP] tag to show this is ready to be reviewed.

Did that, thanks!

**Comment by 9037568** [comment]
Please also review the READMEs provided by other bindings and implement a "Full Example" section in yours.

**Comment by 9037568** [comment]
I'd also like to ask why you've implemented a new toon binding when one already exists.
Is there a reason we shouldn't just update the existing one?

**Comment by DaanMeijer** [comment]
> I'd also like to ask why you've implemented a new toon binding when one already exists.
> Is there a reason we shouldn't just update the existing one?

This is a very different use case. Instead of using the (cloud) API provided by the distributor, which is dodgy at best, this binding allows for using a direct connection, without using the internet.The channels don't map cleanly either.

File: bundles/org.openhab.binding.rootedtoon/src/main/resources/ESH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+			<channel id="BoilerSetpoint" typeId="boilerSetpoint" />
+
+		</channels>
+
+		<config-description>
+			<parameter name="url" type="text" required="true" pattern="https?://.*">
+				<label>URL</label>
+				<description>The URL on which to contact your local Toon.</description>
+			</parameter>
+			<parameter name="refreshInterval" type="integer" required="true" min="1" max="3600">
```
[the remark this answers is not in the document]
Comment (DaanMeijer) [reply bundles/org.openhab.binding.rootedtoon/src/main/resources/ESH-INF/thing/thing-types.xml:77]: The device updates only once every 8 seconds. However, if the polling interval is 8 seconds as well, there might be a lag between the device refreshing and openHAB noticing those changes. It is up to the user to decide whether they would rather fire more requests, or have a bit of lag.

**Comment by fwolter** [comment]
Could you fix the build errors? I would take a look into it, then.

**Comment by DaanMeijer** [comment]
I would like to apologize to all the code authors I apparently requested a review from. My git mishaps caused github to ask you to review my code, but this wasn't my intention. Again, I'm sorry.

**Pull request closed without merging**